### PR TITLE
Added Psalm to CI pipeline

### PR DESCRIPTION
**Pull request by Skyld**
- Added Psalm (https://psalm.dev/) static code analyser to CI pipeline
- Fixed some code issues

No need for changelog or release.

### CODE REVIEW

File: psalm.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
+<psalm
+        autoloader="bin/.phpunit/phpunit-8.5-0/vendor/autoload.php"
+        errorLevel="3"
+        reportMixedIssues="false"
```
Comment (andreluizmachado) [inline psalm.xml:5]: Can we enable it?

File: psalm.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
```
Comment (andreluizmachado) [inline psalm.xml:1]: Can we rename this file to psalm.xml.dist?

**Review by andreluizmachado — COMMENTED** [review]
Hi. I just added two comments.

File: .env [hunk 3]
```diff
@@ -19,8 +19,7 @@ JWT_REFRESH_TOKEN_TTL=86400
 # IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
 #
 # DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
-# DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7"
-# DATABASE_URL="postgresql://db_user:db_password@127.0.0.1:5432/db_name?serverVersion=13&charset=utf8"
```
Comment (uncleempty) [inline .env:23]: just to check: why pgsql example was removed?

File: tests/bootstrap.php [hunk 4]
```diff
@@ -24,8 +24,6 @@
 
 require dirname(__DIR__) . '/vendor/autoload.php';
 
-if (file_exists(dirname(__DIR__) . '/config/bootstrap.php')) {
-    require dirname(__DIR__) . '/config/bootstrap.php';
-} elseif (method_exists(Dotenv::class, 'bootEnv')) {
```
Comment (uncleempty) [inline tests/bootstrap.php:29]: just to check, and what if method doesn't exist -> just fail? no need to require boostrap file?

**Review by uncleempty — COMMENTED** [review]
Please check comments

File: psalm.xml (hunk 1 above)
Comment (Skyld) [reply psalm.xml:5]: Done :)

File: psalm.xml (hunk 2 above)
Comment (Skyld) [reply psalm.xml:1]: Done!

File: .env (hunk 3 above)
Comment (Skyld) [reply .env:23]: Mistake of symfony recipe autoupdate, reverted!

File: tests/bootstrap.php (hunk 4 above)
Comment (Skyld) [reply tests/bootstrap.php:29]: That file doesn't not exist anymore in Symfony 5. About the method check, I removed the check, you are right, it cannot work without it.

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Changelog is updated according to changes (if applicable)
- [x] Documentation is updated according to changes (if applicable)

File: src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php [hunk 5]
```diff
@@ -316,7 +316,7 @@ private function initializeDate(string $dateString): ?DateTime
         $dateObj = null;
         if (!empty($dateString)) {
             try {
-                $date = Carbon::createFromFormat(Carbon::ISO8601, $dateString);
+                $date = Carbon::createFromFormat(Carbon::ATOM, $dateString);
```
Comment (luisfn) [inline src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php:319]: 🤦‍♂️

File: psalm.xml.dist [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<psalm
+        autoloader="bin/.phpunit/phpunit-8.5-0/vendor/autoload.php"
```
Comment (luisfn) [inline psalm.xml.dist:3]: Not sure why but I had to change this to `bin/.phpunit/phpunit-8.3-0/vendor/autoload.php` on my local, even with fresh install

**Review by luisfn — APPROVED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**